identity_credential/src/validator/vc_jwt_validation/credential_jwt_validator.rs

```diff
@@ -0,0 +1,546 @@
+    if status_check == crate::validator::StatusCheck::SkipAll {
+      return Ok(());
+    }
+
+    match &credential.credential_status {
+      None => Ok(()),
+      Some(status) => {
+        // Check status is supported.
+        if status.type_ != crate::revocation::RevocationBitmap::TYPE {
+          if status_check == crate::validator::StatusCheck::SkipUnsupported {
+            return Ok(());
+          }
+          return Err(ValidationError::InvalidStatus(crate::Error::InvalidStatus(format!(
+            "unsupported type '{}'",
+            status.type_
+          ))));
+        }
+        let status: crate::credential::RevocationBitmapStatus =
+          crate::credential::RevocationBitmapStatus::try_from(status.clone())
+            .map_err(ValidationError::InvalidStatus)?;
+
+        // Check the credential index against the issuer's DID Document.
+        let issuer_did: CoreDID = Self::extract_issuer(credential)?;
+        trusted_issuers
+          .iter()
+          .find(|issuer| <CoreDocument>::id(issuer.as_ref()) == &issuer_did)
+          .ok_or(ValidationError::DocumentMismatch(SignerContext::Issuer))
+          .and_then(|issuer| Self::check_revocation_bitmap_status(issuer, status))
+      }
+    }
+  }
+
+  /// Check the given `status` against the matching [`RevocationBitmap`] service in the
+  /// issuer's DID Document.
+  #[cfg(feature = "revocation-bitmap")]
+  fn check_revocation_bitmap_status<DOC: AsRef<CoreDocument> + ?Sized>(
+    issuer: &DOC,
+    status: crate::credential::RevocationBitmapStatus,
+  ) -> ValidationUnitResult {
+    use crate::revocation::RevocationDocumentExt;
+
+    let issuer_service_url: identity_did::DIDUrl = status.id().map_err(ValidationError::InvalidStatus)?;
+
+    // Check whether index is revoked.
+    let revocation_bitmap: crate::revocation::RevocationBitmap = issuer
+      .as_ref()
+      .resolve_revocation_bitmap(issuer_service_url.into())
+      .map_err(|_| ValidationError::ServiceLookupError)?;
+    let index: u32 = status.index().map_err(ValidationError::InvalidStatus)?;
+    if revocation_bitmap.is_revoked(index) {
+      Err(ValidationError::Revoked)
+    } else {
+      Ok(())
+    }
+  }
+
+  /// Utility for extracting the issuer field of a [`Credential`] as a DID.
+  ///
+  /// # Errors
+  ///
+  /// Fails if the issuer field is not a valid DID.
+  pub fn extract_issuer<D, T>(credential: &Credential<T>) -> std::result::Result<D, ValidationError>
+  where
+    D: DID,
+    <D as FromStr>::Err: std::error::Error + Send + Sync + 'static,
+  {
+    D::from_str(credential.issuer.url().as_str()).map_err(|err| ValidationError::SignerUrl {
+      signer_ctx: SignerContext::Issuer,
+      source: err.into(),
+    })
+  }
+}
+
+impl Default for CredentialValidator {
+  fn default() -> Self {
+    Self::new()
+  }
+}
+
+#[cfg(test)]
+mod tests {
+  use identity_core::common::Duration;
+  // All tests here are essentially adaptations of the old CredentialValidator tests.
```

Comment:
There are still many left to adapt though. It may make sense to spread the testing of this type more out. In particular if we take a stance that signing should only be done using a `Storage` there can be some tests here where we mock the signature (and possibly the verifier)  and then some proper integration tests in `identity_storage`.